datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1222,9 +1275,10 @@ public ZkBackedLiveInstanceListProvider() {
       List<String> liveInstances = new ArrayList<>();
       for (String n : nodes) {
         String hostname = _zkclient.ensureReadData(KeyBuilder.liveInstance(_cluster, n));
-        if (hostname != null) {
+        if (hostname == null) {
```

Comment:
I remember reading this at some point and thinking I am going to see this in this the logs somewhere and assume something bad happened, and that exactly happened yesterday. Awesome that you fixed it.